packages/actor-rdf-parse-html/lib/ActorRdfParseHtml.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+} from "@comunica/bus-rdf-parse";
+import {
+  IActionRdfParseHtml,
+  IActorRdfParseHtmlOutput,
+  IHtmlParseListener,
+} from "@comunica/bus-rdf-parse-html";
+import {ActionContext, Actor, Bus, IActorTest} from "@comunica/core";
+import {Parser as HtmlParser} from "htmlparser2";
+import * as RDF from "rdf-js";
+import {Readable} from "stream";
+
+/**
+ * A comunica HTML RDF Parse Actor.
+ * It creates an HTML parser, and delegates its events via the bus-rdf-parse-html bus to other HTML parsing actors.
+ */
+export class ActorRdfParseHtml extends ActorRdfParseFixedMediaTypes {
+
+  private readonly busRdfParseHtml: Bus<Actor<IActionRdfParseHtml, IActorTest,
+    IActorRdfParseHtmlOutput>, IActionRdfParseHtml, IActorTest, IActorRdfParseHtmlOutput>;
+
+  constructor(args: IActorRdfParseHtmlArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+    const quads = new Readable({ objectMode: true });
+    quads._read = async () => {
+      // Only initialize once
+      quads._read = null;
+
+      // Create callbacks action
+      let endBarrier = 1;
+      const emit = (quad: RDF.Quad) => quads.emit('data', quad);
+      const error = (e: Error) => quads.emit('error', e);
+      const end = () => {
+        if (--endBarrier === 0) {
+          quads.push(null);
+        }
+      };
+      const htmlAction = {
+        baseIRI: action.baseIRI,
+        context,
+        emit,
+        end,
+        error,
+        headers: action.headers,
+      };
+
+      // Register html parse listeners
+      Promise.all(this.busRdfParseHtml.publish(htmlAction))
```

Comment:
Why does this need multiple listeners? Do all of them parse a specific html part?

Reply:
This is done like this to make sure that only a single HTML parser is needed. Each listener in from bus then simply listens to the event of the HTML parser. Saves us quite a bit of CPU cycles when 1+ listeners are present.